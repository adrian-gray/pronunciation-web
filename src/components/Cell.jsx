import React from 'react'
import SplitHilite from './SplitHilite'

import {
  TableCell,
  Typography,
  withStyles
} from '@material-ui/core'

const styles = (theme) => ({
  cell: theme.cell,
  largeText: theme.largeText
})

class Cell extends React.Component {
  constructor (props) {
    super(props)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick (e) {
    e.preventDefault()
    this.props.handleClick({
      column: this.props.column,
      row: this.props.row,
      str: this.props.str
    })
  }

  render () {
    const { classes, hilite, key, str } = this.props
    const cellClasses = hilite ? `${classes.cell} ${hilite}` : classes.cell

    return (
      <TableCell className={cellClasses} key={key} onClick={this.handleClick}>
        <Typography>
          <SplitHilite str={str} />
        </Typography>
      </TableCell>
    )
  }
}

export default withStyles(styles)(Cell)
