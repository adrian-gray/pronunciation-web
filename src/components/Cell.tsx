import React from "react";
import SplitHilite from "./SplitHilite";

export default props => {
  const handleClick = e => {
    e.preventDefault();
    if (!props.handleClick) return;
    props.handleClick({
      column: props.column,
      row: props.row,
      str: props.str
    });
  };

  const { hilite, str } = props;
  const cellClasses = hilite ? `cell hilite` : "cell";

  return (
    <td className={cellClasses} key={props.column} onClick={handleClick}>
      <p>
        <SplitHilite str={str} />
      </p>
    </td>
  );
};
