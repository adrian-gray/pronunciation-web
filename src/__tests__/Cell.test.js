import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import Cell from "./../components/Cell";

configure({ adapter: new Adapter() });
afterEach(cleanup);

test("Cell", async () => {
  const hilite = true;
  const handleClick = function noop() {};
  const { container } = render(
    <table>
      <tbody>
        <tr>
          <Cell
            hilite={hilite}
            key="dog"
            str="rabbit"
            handleClick={handleClick}
          />
        </tr>
      </tbody>
    </table>
  );

  expect(container.firstChild).toMatchSnapshot();
});

test("Cell with hilite", async () => {
  const hilite = false;
  const { container } = render(
    <table>
      <tbody>
        <tr>
          <Cell hilite={hilite} key="dog" str="rabbit" handleClick={null} />
        </tr>
      </tbody>
    </table>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("Cell click callback", async () => {
  const hilite = true;
  const mockCallback = jest.fn(x => x);
  const container = shallow(
    <table>
      <tbody>
        <tr>
          <Cell
            hilite={hilite}
            key="dog"
            str="rabbit"
            handleClick={mockCallback}
          />
        </tr>
      </tbody>
    </table>
  );

  const cell = container.find("tr");
  expect(cell.length).toEqual(1);
  cell.simulate("click");
  expect(mockCallback.mock.calls.length).toBe(1);
});
