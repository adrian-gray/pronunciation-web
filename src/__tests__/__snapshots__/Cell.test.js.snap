// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="MuiTableCell-root makeStyles-cell-1 true"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          <span>
            rabbit
          </span>
           
        </p>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Cell with hilite 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="MuiTableCell-root makeStyles-cell-45"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          <span>
            rabbit
          </span>
           
        </p>
      </td>
    </tr>
  </tbody>
</table>
`;
